sdk/storage/Azure.Storage.Queues/src/QueueClient.cs

```diff
@@ -2253,6 +2508,27 @@ private void SetNameFieldsIfNull()
             }
         }
         #endregion UpdateMessage
+
+        #region Encoding
+        private static BinaryData? ToBinaryData(string input)
+        {
+            if (input == null)
+            {
+                return default;
+            } else
+            {
+                return new BinaryData(input);
+            }
+        }
+
+        private void AssertEncodingForEncryption()
+        {
+            if (_messageEncoding != QueueMessageEncoding.UTF8)
+            {
+                throw new ArgumentException("Message encoding must be None if encryption is enabled");
+            }
+        }
```

Comment:
There is no "None" value as far as I can tell. Only `UTF8` and `Base64`. Also why can't we encrypt raw bytes?

Reply:
@jaschrep-msft That `None` got forgotten while renaming... will fix.
I think the wording of this error message and/or QueueClientOptions might be not best. We sure can encrypt raw bytes. The point is that encryption already makes each message Base64 string when transmiting over the wire so having both encryption and Base64 here would lead to double Base64 encoding.

Reply:
Perhaps we should be validating client options on construction then? The error message could say something like if you use clientside encryption, then you don't get to make the specification because the encryption module handles it for you.

Reply:
I like that idea.

Reply:
Unfortunatelly QueueClientOptions is mutable and there are code paths (extension) that just clone internal state of service client to queue client plus encryption options, so best I could do is to move this assertion to QueueClient constructors.